test(color-picker): add component token E2E tests

jcfranco opened the pull request:
**Related Issue:** #7180

## Summary

✨🧪✨

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+          shadowSelector: `.${CSS.colorModes}`,
+          targetProp: "--calcite-tab-nav-indicator-color",
+        },
+        "--calcite-color-picker-tab-title-background-color": {
```
I don't think this token is actually effecting anything.

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+            targetProp: "--calcite-color-picker-hex-input-border-color",
+          },
+        ],
+        "--calcite-color-picker-input-corner-radius": [
```
I don't think this token is actually effecting anything.

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+            targetProp: "--calcite-color-picker-hex-input-background-color",
+          },
+        ],
+        "--calcite-color-picker-input-background-color-hover": [
```
I don't think this token is actually effecting anything.

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+            state: "hover",
+          },
+        ],
+        "--calcite-color-picker-button-text-color": [
```
--calcite-color-picker-text-color and --calcite-color-picker-button-text-color have the same color by default. I think we can simplify this to 
`--calcite-color-picker-text-color` + the various states

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+            state: "hover",
+          },
+        ],
+        "--calcite-color-picker-button-icon-color": [
```
none of the -button-icon-color(-) tokens are used. Remove.

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+            targetProp: "--calcite-color-picker-hex-input-prefix-text-color",
+          },
+        ],
+        "--calcite-color-picker-input-suffix-background-color": [
```
pefix/suffix tokens aren't used.

alisonailea commented on packages/calcite-components/src/components/color-picker/color-picker.e2e.ts:
```diff
@@ -2345,4 +2355,274 @@ describe("calcite-color-picker", () => {
+          shadowSelector: `.${CSS.channel}[data-channel-index="3"]`,
+          targetProp: "--calcite-input-suffix-background-color",
+        },
+        "--calcite-color-picker-input-suffix-text-color": {
```
this doesn't effect anything. remove?

alisonailea reviewed (commented):
I opened the tests in debugger mode and it looks like several of the tokens are not applied?

jcfranco commented:
@alisonailea I've removed some of the unused props you pointed out. Thanks for the catch! ✨🎣✨

Others that were pointed out were working, but not shown up for specific configurations or because there was an issue with styles. I've also fixed these and here's how they're looking:

<img width="355" alt="Screenshot 2024-07-16 at 3 09 15 PM" src="https://github.com/user-attachments/assets/30585302-66a7-4e9c-84bc-b76201d07e20">

A couple of questions on ☝️:

1. the background/text color of prefix and suffixes should be the same, so WDYT about consolidating these to `--calcite-color-picker-affix-background-color` and `--calcite-color-picker-affix-text-color`?
2. both button icon/text color props can be applied, but both can be applied due to specificity and `currentColor`: should we drop the icon prop in favor of the button-text prop or, following your suggestion, rely on the component's text color prop? The buttons do change background color when hovered, so it might be useful to allow customizing this differently from the component's text color.

alisonailea commented:
1. yes I think `--calcite-color-picker-affix-text-color` makes sense
2. I'm all for dropping extraneous tokens

jcfranco commented:
@alisonailea Updated. Would you mind reviewing this again? 🙏
